statsmodels/genmod/generalized_linear_model.py

```diff
@@ -41,7 +41,13 @@
 
 
 def _check_convergence(criterion, iteration, tol):
-    return not (np.fabs(criterion[iteration] - criterion[iteration-1]) > tol)
+    if iteration == 0:
+        return False
```

Comment:
Maybe I'm confused, but the 0th element of `criterion[iteration]` is set to be `np.inf`. The first iteration will compare `np.inf` vs. the deviance after the first IRLS iteration. So, it shouldn't converge ever. 

So I think you are right, we don't need the `iteration == 0` case as `np.inf - x` should always be > tol.

No?